src/spikeinterface/core/template.py

```diff
@@ -0,0 +1,196 @@
+import numpy as np
+import json
+from dataclasses import dataclass, field, astuple
+from .sparsity import ChannelSparsity
+
+
+@dataclass
+class Templates:
+    """
+    A class to represent spike templates, which can be either dense or sparse.
+
+    It is constructed with the following parameters:
+    ----------
+    templates_array : np.ndarray
+        Array containing the templates data.
+    sampling_frequency : float
+        Sampling frequency of the templates.
+    nbefore : int
+        Number of samples before the spike peak.
+    sparsity_mask : np.ndarray, optional (default=None)
```

Comment:
```suggestion
    sparsity_mask : np.ndarray or None, default: None
```

I think the notation decided on was this for the docstring (whereas the current is pure Numpy Docstring format. But I would let @alejoe91 confirm the format before I throw a bunch of edits.

Reply:
Dude, I keep forgetting this. Thanks.